concepts/50 React Components/45 Optimize Performance/05 Do not Pass Inline Functions and Objects as Component Props/10 Functions Configuration.md

```diff
@@ -0,0 +1,53 @@
+If the in-line handlers do not require an access to the component variables, keep them out of the component body. Otherwise, keep them out of the render function (if you use class components) or wrap them in the [useCallback](https://reactjs.org/docs/hooks-reference.html#usecallback) hook (if you use functions).
+
+    <!-- tab: App.js -->// Incorrect:
+    function App() {
+        // ...
+        return (
+            <DataGrid
+                id="grid-container"
+                dataSource={employees}
+                selectedRowKeys={keys}
+                keyExpr="ID"
+                onCellPrepared={(e) => {
+                    if (e.row && e.row.isSelected) {
+                        e.cellElement.style.backgroundColor = "green";
+                        console.log("re-render");
+                    }
+                }}
+                showBorders={true}
+            >
+            </DataGrid>
+        );
+    }
+
+    export default App;
+
+    // Correct:
```

Comment:
We should display both cases: the useCallback and extracted function. Use comments to convey what's what.